diksha/diksha.controller.js

```diff
@@ -492,80 +493,84 @@ let getHomePage = (req, res) => {
-                } else {
-                    queryString += filters[key] + (' ');
+            let configFilters = {};
+	        let bulkPromises = [];
+            let sectionResponsePromises = [];
+            let sections = loadedJson.response.sections;
+            let sectionResponse = [];
+            for (let i in sections) {
+		        let sectionObject= {};
+                sectionNames[i] = sections[i].display.name.en;
+		        configFilters = sections[i].searchQuery.request.filters;
+                for (let key in filters) {
+                    sectionObject[key] = filters[key];
                 }
-            }
-            query.query = queryString;
-            return doPrebuiltSearch(prebuiltQueryStructures, queryString);
-        }).then(value => {
-            let responses = value.responses;
-            genieResponses = responses['Best of Genie'];
-
-            //	fs.writeFile("/home/admin/Final_thingy", JSON.stringify(genieResponses, null, 4), (err, res) => {});
-
-            return loadSkeletonJson('homePageResponseSkeleton');
-        }).then(value => {
-            responseStructure = value.data;
-            return doThoroughSearch(query);
-        }).then(value => {
-            let responses = value.responses;
-
-            //	console.log("_____HERE____");
-            //	console.log({query});
-            //      console.log({responses});
-
-            return generateResponseStructure(responseStructure, responses);
-        }).then(value => {
-            responseStructure = value.responseStructure;
-            //responseStructure.result.page.sections[i].contents = responses;
-            responseStructure.result.response.sections[0].contents = genieResponses;
-            responseStructure.ts = new Date();
-            responseStructure.ver = parsedReq.ver;
-            responseStructure.id = parsedReq.id;
-            responseStructure.name = parsedReq.request.name;
-            responseStructure.resmsgid = '0211201a-c91e-41d6-ad25-392de813124c';
-
-            //console.log(JSON.stringify(responseStructure, null, 4));
-            //fs.writeFile("/home/admin/api.debug", JSON.stringify(responseStructure), (err, res) => console.log('Written debug info to api.debug'));
+                for (let key in configFilters) {
+                    sectionObject[key] = configFilters[key];  
+                }
+                //console.log({sectionObject});
+	            sectionResponsePromises.push(doSectionwiseSearch(sectionObject));
+            }   
+		return q.all(sectionResponsePromises);				
+	}).then(value => {
+		let responsePromises = [];
+        for(let i in value) {
+            responsePromises.push(getAllPromises(value[i]));
+        }
+        return q.all(responsePromises); 
+	}).then(value => {
+		let responses = {};
+		let defer = q.defer();
+        for(let i in  sectionNames) {
+            responses[i] = value[i].responses.map(response => response.fields)
+        }
+        return responses;	
```

Comment:
Why return `responses` and put the rest of the statements under a separate then?

Reply:
this is done just to map the appropriate response to its correct section and then passed on to homepage to push it

Reply:
`responses` are not a promises, right? They are resolved values. So after having map-ped them, why do you return and start another then block to perform further operations? Statements following this one be put under the same function.

Reply:
inorder to know that what all is happening, i have separated the sections and put them in a then ladder so that in future we know what and how are we making a query, mapping the response and finally pushing it to be displayed on HomePage